apps/src/lib/node/ledger/shell/process_proposal.rs

```diff
@@ -67,116 +87,183 @@ where
+                    }
+                })
+            },
+        );
+        let tx = match maybe_tx {
+            Ok(tx) => tx,
+            Err(tx_result) => return tx_result,
         };
+
         // TODO: This should not be hardcoded
         let privkey = <EllipticCurve as PairingEngine>::G2Affine::prime_subgroup_generator();
 
-        match process_tx(tx) {
-            // This occurs if the wrapper / protocol tx signature is invalid
-            Err(err) => TxResult {
-                code: ErrorCodes::InvalidSig.into(),
-                info: err.to_string(),
+        match tx {
+            // If it is a raw transaction, we do no further validation
+            TxType::Raw(_) => TxResult {
+                code: ErrorCodes::InvalidTx.into(),
+                info: "Transaction rejected: Non-encrypted transactions are \
+                       not supported"
+                    .into(),
             },
-            Ok(result) => match result {
-                // If it is a raw transaction, we do no further validation
-                TxType::Raw(_) => TxResult {
-                    code: ErrorCodes::InvalidTx.into(),
-                    info: "Transaction rejected: Non-encrypted transactions \
-                           are not supported"
-                        .into(),
-                },
-                TxType::Protocol(protocol_tx) => match protocol_tx.tx {
-                    ProtocolTxType::EthereumEvents(_) => TxResult {
-                        code: ErrorCodes::Ok.into(),
-                        info: "Process Proposal accepted this transaction"
-                            .into(),
-                    },
-                    _ => TxResult {
-                        code: ErrorCodes::InvalidTx.into(),
-                        info: "Unsupported protocol transaction type".into(),
-                    },
-                },
-                TxType::Decrypted(tx) => match self.next_wrapper() {
-                    Some(wrapper) => {
-                        if wrapper.tx_hash != tx.hash_commitment() {
-                            TxResult {
-                                code: ErrorCodes::InvalidOrder.into(),
-                                info: "Process proposal rejected a decrypted \
-                                       transaction that violated the tx order \
-                                       determined in the previous block"
-                                    .into(),
-                            }
-                        } else if verify_decrypted_correctly(&tx, privkey) {
+            TxType::Protocol(protocol_tx) => match protocol_tx.tx {
+                ProtocolTxType::EthereumEvents(digest) => {
+                    *vote_ext_digest_num += 1;
+
+                    let extensions =
+                        digest.decompress(self.storage.last_height);
+                    let filtered_extensions = self
+                        .filter_invalid_vote_extensions_residuals(extensions);
```

Comment:
similarly here we are not filtering anything, we are getting the validation result for each vote extension we have. (each result including some useful things like the signed extension and the voting power if the validation did pass).